src/silx/gui/plot/ImageStack.py

```diff
@@ -145,49 +147,90 @@ def _urlChanged(self, value):
-        super(UrlList, self).__init__(parent)
-        self.setLayout(qt.QVBoxLayout())
-        self.layout().setSpacing(0)
-        self.layout().setContentsMargins(0, 0, 0, 0)
-        self._listWidget = qt.QListWidget(parent=self)
-        self.layout().addWidget(self._listWidget)
+        super().__init__(parent)
+        self._editable = False
+        # are we in 'editable' mode: for now if true then we can remove some item from the list
+
+        # menu to be triggered when in edition from right-click
+        self._menu = qt.QMenu()
+        self._removeAction = qt.QAction(
+            text="Remove",
+            parent=self
+        )
+        self._removeAction.setShortcuts(
+            [
+                # qt.Qt.Key_Delete,
+                qt.QKeySequence.Delete,
+            ]
+        )
+        self._menu.addAction(self._removeAction)
 
         # connect signal / Slot
-        self._listWidget.currentItemChanged.connect(self._notifyCurrentUrlChanged)
-
-        # expose API
-        self.currentItem = self._listWidget.currentItem
+        self.currentItemChanged.connect(self._notifyCurrentUrlChanged)
+
+    def setEditable(self, editable: bool):
+        if editable != self._editable:
+            self._editable = editable
+            # discusable choice: should we change the selection mode ? No much meaning
+            # to be in ExtendedSelection if we are not in editable mode. But does it has more
+            # meaning to change the selection mode ?
+            if editable:
+                self._removeAction.triggered.connect(self._removeSelectedItems)
+                self.addAction(self._removeAction)
+                self.setSelectionMode(qt.QAbstractItemView.ExtendedSelection)
+            else:
+                self._removeAction.triggered.disconnect(self._removeSelectedItems)
+                self.removeAction(self._removeAction)
+                self.setSelectionMode(qt.QAbstractItemView.SingleSelection)
 
     def setUrls(self, urls: list) -> None:
         url_names = []
         [url_names.append(url.path()) for url in urls]
-        self._listWidget.addItems(url_names)
+        self.addItems(url_names)
 
     def _notifyCurrentUrlChanged(self, current, previous):
         if current is None:
-            pass
+            self.sigCurrentUrlChanged.emit("")
         else:
             self.sigCurrentUrlChanged.emit(current.text())
 
-    def setUrl(self, url: DataUrl) -> None:
-        assert isinstance(url, DataUrl)
-        sel_items = self._listWidget.findItems(url.path(), qt.Qt.MatchExactly)
-        if sel_items is None:
-            _logger.warning(url.path(), ' is not registered in the list.')
-        elif len(sel_items) > 0:
-            item = sel_items[0]
-            self._listWidget.setCurrentItem(item)
-            self.sigCurrentUrlChanged.emit(item.text())
-
-    def clear(self):
-        self._listWidget.clear()
+    def setUrl(self, url: typing.Optional[DataUrl]) -> None:
```

Comment:
```suggestion
    def setUrl(self, url: typing.Optional[DataUrl]) -> None:
    """Set the current URL.

    :param url: The new selected URL. Use `None` to clear the selection.
    """
```

What about renaming this to `setCurrentUrl` and also provide a `setUrl` with a deprecation warning?